internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -116,6 +117,13 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 	}
 
+	if reconcileManagedLabels(machine, node) {
```

Comment:
I assume this results in only the labels being set with SSA? (looks okay, just not 100% sure)

I wonder if we should just write both labels and annotations with SSA with just one call. Could be simpler